login1/dbus.go

```diff
@@ -155,6 +156,72 @@ func userFromInterfaces(user []interface{}) (*User, error) {
 	return &ret, nil
 }
 
+//GetActiveSession may be used to get the session object path for the current active session
+func (c *Conn) GetActiveSession() (dbus.ObjectPath, error) {
+	var activeSessionPath dbus.ObjectPath
+	var seat0Path dbus.ObjectPath
+	err := c.object.Call(dbusInterface+".GetSeat", 0, "seat0").Store(&seat0Path)
+
+	if err != nil {
+		return activeSessionPath, err
+	}
+
+	seat0Obj := c.conn.Object(dbusDest, seat0Path)
+	activeSession, err := seat0Obj.GetProperty(dbusDest + ".Seat.ActiveSession")
+	if err != nil {
+		return activeSessionPath, err
+	}
+
+	activeSessionPath = activeSession.Value().([]interface{})[1].(dbus.ObjectPath)
```

Comment:
This can panic, which is not great. It can be improved by checking that:
 * the array has at least an entry
 * the typecast did not fail